harpy/_validations.py

```
@@ -257,17 +257,32 @@ def validate_popsamples(infiles, popfile, quiet):
+                    click.echo(segment_id)
+                    sys.exit(1)
+                else:
+                    segment_ids.add(segment_id)
+            except ValueError:
+                # skip rows without two columns
+                continue
+    if not code_letters:
+        print_error("incorrect schema format", f"Schema file {os.path.basename(infile)} has no valid rows. Rows should be sample<tab>segment, e.g. sample_01<tab>C75")
+        sys.exit(1)
+    if len(code_letters) > 1:
+        print("invalid schema", f"Schema file {os.path.basename(file_path)} has sample IDs expected to be identified across multiple barcode segments. All sample IDs for this technology should be in a single segment, such as [bold green]C[/bold green] or [bold green]D[/bold green].")
```

Ruff (0.8.2)

Undefined name `file_path`

(F821)